src/mcts/utils.h

```diff
@@ -226,7 +268,31 @@ class MctsResult {
   }
 
   // assume already normalized
-  void sample() {
+  void sampleWithoutSmoothing() {
+    auto weight = [this](float pival) {
```

Comment:
- Why is there an identity lambda?
- Why does it capture `[this]`?


Suggestion: remove entirely

Reply:
the issue is that sampleDiscreteProbability takes in, among other things, a function which is used to weight the values. To fit this paradigm, I just deleted all the transformations in the original function, and left it as an identity.

Is there a better way to write the identity function?

Reply:
Yeah but a lambda is created there anyways. You can just remove `weight()` from line 291.

Reply:
got it, thanks